src/robot/api/deco.py

```diff
@@ -41,8 +53,28 @@ def exposed_as_keyword():
 not_keyword.robot_not_keyword = True
 
 
+# Decorator used without arguments, i.e. @keyword()
+@overload
+def keyword(name: F) -> F:
+    ...
+
+
+# Decorator used with arguments, e.g. @keyword("Custom Keyword Name")
+@overload
+def keyword(
+    name: 'str | None' = None,
+    tags: Sequence[str] = (),
+    types: 'Sequence[str] | Mapping[str, Any]' = (),
```

Comment:
This typing for types is wrong. Types can be specified using actual types, strings and tuples. Probably using union types created with `x | y` would work as well. And when types are given as a sequence, using `None` means that that argument doesn't have a type. Finally, using `types=None` means that argument conversion should be disabled altogether.

I guess there are three options:
1. Don't add typing to types at all. Just use `types: 'Seqeunce|Mapping|None' = ()`. This is correct but not very detailed.
2. Created a type alias like `TypeHint` that represents all possible ways a type can be specified properly. Then we ought to be able to use `types: Sequence[TypeHint|None]|Mapping[str, TypeHint]|None = ()`. This ought to be correct and detailed, but it is also rather complicated.
3. Same as 2. but instead of creating type aliases here, reuse type aliases from the `interfaces` module. It has both `TypeHint` for an individual type and `TypeHints` that covers Sequence and Mapping cases. With the latter we could reduce this to `types 'TypeHints|None' = ()` which looks pretty good.

I believe the option 3. is a clear winner here. Notice that at the moment `interfaces` has `Type` and `Types` instead of `TypeHint` and `TypeHints`, but I'll rename them to avoid confusion with `typing.Type`. There are also some other things that need to be fixed and I'll do that at the same time. That will likely cause conflict with this PR, but I try to fix the conflict afterwards.

Reply:
I agree that 3. is preferred here. I would propose adding a `robot.annotations` module and move the `TypeHints` and other long / complex annotations there.

That annotations module would then be a place to look for "complex types / interfaces / Protocols" that are common within the robot codebase. I'm thinking of the annotations we'd see in e.g. the listener interface also. Extra documentation on the why of certain signatures could also be centralized then. In addition, it may prevent circular import issues.

Would you support adding such an annotations module?

Reply:
Having a separate module for annotations is fine in general, but I'm not sure do we have enough needs for it yet. If we only need to reuse `TypeHints`, this module importing it from `interfaces` is simpler. I believe we should get this PR merged and look st how many reusable types we have after that.

Reply:
Changed to option 3.